simd/src/Kokkos_Simd_Common.hpp

```diff
@@ -0,0 +1,224 @@
+#ifndef KOKKOS_SIMD_COMMON_HPP
+#define KOKKOS_SIMD_COMMON_HPP
+
+#include <cmath>
+#include <cstring>
+
+#include <Kokkos_Macros.hpp>
+
+#include <desul/atomics/Common.hpp>
+
+namespace Kokkos {
+namespace Experimental {
+
+template <class To, class From>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr To bit_cast(
+    From const& src) {
+  To dst;
+  std::memcpy(&dst, &src, sizeof(To));
+  return dst;
+}
+
+template <class T, class Abi>
+class simd;
+
+template <class T, class Abi>
+class simd_mask;
+
+class element_aligned_tag {};
+
+template <class Mask, class Value>
+class const_where_expression {
+ protected:
+  Value& m_value;
+  Mask const& m_mask;
+
+ public:
+  const_where_expression(Mask const& mask_arg, Value const& value_arg)
+      : m_value(const_cast<Value&>(value_arg)), m_mask(mask_arg) {}
+};
+
+template <class Mask, class Value>
+class where_expression : public const_where_expression<Mask, Value> {
+  using base_type = const_where_expression<Mask, Value>;
+
+ public:
+  where_expression(Mask const& mask_arg, Value& value_arg)
+      : base_type(mask_arg, value_arg) {}
+};
+
+template <class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION where_expression<simd_mask<T, Abi>,
+                                                           simd<T, Abi>>
+where(desul::Impl::dont_deduce_this_parameter_t<simd_mask<T, Abi>> const& mask,
+      simd<T, Abi>& value) {
+  return where_expression(mask, value);
+}
+
+template <class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION const_where_expression<
+    simd_mask<T, Abi>, simd<T, Abi>>
+where(desul::Impl::dont_deduce_this_parameter_t<simd_mask<T, Abi>> const& mask,
```

Comment:
No, do not use implementation details from `desul`.  I am aware that they are a view occurrences in Kokkos but let's not introduce more violations.
Either define your own, or use `Impl::identity<T>` from `<impl/Kokkos_Utilities.hpp>`.  I really dislike the name but that is essentially a backport from the C++20 facility `std::type_identity`.

Reply:
Okay, will switch to `Impl::identity<T>`, thanks!